processing/src/main/java/org/apache/druid/segment/join/Joinables.java

```diff
@@ -114,4 +118,37 @@ public static boolean isPrefixedBy(final String columnName, final String prefix)
       );
     }).collect(Collectors.toList());
   }
+
+  public static void checkPreJoinableClausesForDuplicatesAndShadowing(
+      final List<PreJoinableClause> preJoinableClauses
+  )
+  {
+    List<String> prefixes = new ArrayList<>();
+    for (PreJoinableClause clause : preJoinableClauses) {
+      prefixes.add(clause.getPrefix());
+    }
+
+    checkPrefixesForDuplicatesAndShadowing(prefixes);
```

Comment:
I think this should use a Trie of some sort here so the complexity isn't `n + n^2 * k`

I don't know the best data structure to use here off the top of my head, but I can search. How many clauses is reasonable - if it's just 5, this may not matter in the big picture?

Reply:
I think we'd typically have a small number of clauses (one per JOIN statement in the query), I did some brief benchmarks with a radix tree approach and found it was significantly slower on the 6 prefix test I tried (https://github.com/jon-wei/druid/tree/join_prefix_check2)

I think this is fine for now, I added a comment about the naive approach

Reply:
👍 for the benchmarks. In my experience, sorting can actually be harmful at such small numbers. Just curious if you ran benchmarks to see if sorting would be beneficial here. Again, since it's such small numbers, it probably doesn't matter. Sorry for sending you on a wild goose chase with my comment.